yt_dlp/extractor/ondemandkorea.py

```diff
@@ -1,87 +1,164 @@
-            base_url=url)
-
-        info.update({
-            'title': title,
-            'description': self._og_search_description(webpage),
-            'thumbnail': self._og_search_thumbnail(webpage)
-        })
-        return info
+
+        data = self._download_json(f'https://odkmedia.io/odx/api/v3/playback/{video_id}/', video_id,
+                                   fatal=False, headers={'service-name': 'odk'},
+                                   query={'did': random_uuidv4()}, expected_status=(403, 404))
+        if not data.get('result'):
+            raise ExtractorError(traverse_obj(data, ('messages', '__default'), 'title'), expected=True)
+
+        potential_urls = traverse_obj(data, ('result', ('sources', 'manifest'), ..., 'url'))
+        # Try to bypass geo-restricted ad proxy
+        potential_urls = [
+            alt_url if (alt_url := traverse_obj(url, ({parse_qs}, 'stream_url', 0, {url_or_none}))) else url
+            for url in potential_urls
+        ]
+        # Try to upgrade quality
+        potential_urls = [
+            mod_url if self._request_webpage(
+                HEADRequest(mod_url := re.sub(r'_720(p?)\.m3u8', r'_1080\1.m3u8', url)), video_id,
+                note='Checking if higher quality format is available', fatal=False) else url
+            for url in potential_urls
+        ]
+
+        formats = []
+        for url in potential_urls:
+            formats.extend(self._extract_m3u8_formats(url, video_id, fatal=False))
+
+        subtitles = {}
+        for track in traverse_obj(data, ('result', 'text_tracks', lambda _, v: url_or_none(v['url']))):
+            subtitles.setdefault(track.get('language', 'und'), []).append({
+                'url': track['url'],
+                'ext': track.get('codec'),
+                'name': track.get('label'),
+            })
+
+        return {
+            'id': video_id,
+            'title': join_nonempty(
+                ('result', 'episode', 'program', 'title'),
+                ('result', 'episode', 'title'), from_dict=data, delim=': '),
+            **traverse_obj(data, ('result', {
+                'thumbnail': ('episode', 'images', 'thumbnail', {url_or_none}),
+                'release_date': ('episode', 'release_date', {lambda x: x.replace('-', '')}, {unified_strdate}),
+                'duration': ('duration', {functools.partial(float_or_none, scale=1000)}),
+                'age_limit': ('age_rating', 'name', {lambda x: x.replace('R', '')}, {parse_age_limit}),
+                'series': ('episode', {lambda x: x['program'] if x['kind'] == 'series' else None}, 'title'),
+                'series_id': ('episode', {lambda x: x['program'] if x['kind'] == 'series' else None}, 'id'),
+                'episode': ('episode', {lambda x: x['title'] if x['kind'] == 'series' else None},),
+                'episode_number': ('episode', {lambda x: x['number'] if x['kind'] == 'series' else None}, {int_or_none}),
```

Comment:
maybe a helper would be nice for these, e.g.
```py
def if_series(key=None):
    def func(obj):
        return obj[key] if key and obj['kind'] == 'series' else None
    return func
```
```suggestion
                'series': ('episode', {if_series(key='program')}, 'title'),
                'series_id': ('episode', {if_series(key='program')}, 'id'),
                'episode': ('episode', {if_series(key='title')}),
                'episode_number': ('episode', {if_series(key='number')}, {int_or_none}),
```
and if you don't use this suggestion, then just cleanup the errant comma after the transformation bracket in the path for `episode`